x/msgfees/keeper/proposal_handler.go

```diff
@@ -134,3 +134,12 @@ func HandleUpdateNhashPerUsdMilProposal(ctx sdk.Context, k Keeper, proposal *typ
 	k.SetParams(ctx, params)
 	return nil
 }
+
+// HandleUpdateDenomMetadataProposal handles update of msg fee denom metadata
+func HandleUpdateDenomMetadataProposal(ctx sdk.Context, k Keeper, proposal *types.UpdateDenomMetadataProposal, registry codectypes.InterfaceRegistry) error {
+	if err := sdk.ValidateDenom(proposal.Metadata.Base); err != nil {
+		return err
+	}
```

Comment:
Should use `proposal.Metadata.Validate()` instead of only validating the base denom.